wger/exercises/api/views.py

```diff
@@ -130,6 +130,66 @@ def search(request):
     return Response(json_response)
 
 
+@api_view(['GET'])
+def info(request):
+    """Returns all the information about an exercise."""
+    """
+    It uses the same functionality as search.
+    It takes the exercise name as the parameter
+    and returns all the information about that exercise
+    """
+    q = request.GET.get('term', None)
+    results = []
+    json_response = {}
+
+    if q:
+        languages = load_item_languages(
+            LanguageConfig.SHOW_ITEM_EXERCISES,
+            language_code=request.GET.get("language", None),
+        )
+        exercises = (
+            Exercise.objects.filter(name__icontains=q)
+            .filter(language__in=languages)
+            .filter(status=Exercise.STATUS_ACCEPTED)
+            .order_by("category__name", "name")
+            .distinct()
+        )
+        for exercise in exercises:
+            if exercise.main_image:
+                image_obj = exercise.main_image
+                image = image_obj.image.url
+                t = get_thumbnailer(image_obj.image)
+                thumbnail = t.get_thumbnail(aliases.get("micro_cropped")).url
+            else:
+                image = None
+                thumbnail = None
+
+            exercise_json = {
```

Comment:
Kindly transfer this json object here to some other file and import it here. There are so many lines of code in one method. Consider some refactoring too.

Reply:
@nakatuddesuzan this has been worked on